### PR TITLE
[JENKINS-5253] Allow case insensitive patterns in Artifacts Archiving

### PR DESCRIPTION
**Pull request by dominiquebrice**
Recreating a pull request with a clean base. Nothing to salvage from previous pull-request.

### CODE REVIEW

File: core/src/test/java/hudson/FilePathTest.java [hunk 1]
```diff
@@ -488,6 +489,25 @@ private static void assertValidateAntFileMask(String expected, FilePath d, Strin
                 // good
             }
     }
+    
+    @Issue("5253")
```
Comment (oleg-nenashev) [inline core/src/test/java/hudson/FilePathTest.java]: For the person who merges the PR: Add JENKINS- prefix

File: core/src/main/resources/hudson/tasks/ArtifactArchiver/config.jelly [hunk 2]
```diff
@@ -40,8 +40,11 @@ THE SOFTWARE.
     <f:entry field="fingerprint">
       <f:checkbox title="${%Fingerprint all archived artifacts}"/>
     </f:entry>
-        <f:entry field="defaultExcludes" >
-          <f:checkbox title="${%defaultExcludes}" default="true"/>
-        </f:entry>
+    <f:entry field="defaultExcludes" >
+      <f:checkbox title="${%defaultExcludes}" default="true"/>
+    </f:entry>
+    <f:entry field="caseSensitive" >
```
Comment (oleg-nenashev) [inline core/src/main/resources/hudson/tasks/ArtifactArchiver/config.jelly]: This has been done in the further PR

File: core/src/main/java/hudson/tasks/ArtifactArchiver.java [hunk 3]
```diff
@@ -294,13 +315,19 @@ public String getDisplayName() {
...
+        public FormValidation doCheckArtifacts(@AncestorInPath AbstractProject project,
+                @QueryParameter String value,
+                @QueryParameter(value = "caseSensitive") String caseSensitive)
+                throws IOException {
             if (project == null) {
                 return FormValidation.ok();
             }
-            return FilePath.validateFileMask(project.getSomeWorkspace(),value);
+            // defensive approach to remain case sensitive in doubtful situations
+            boolean bCaseSensitive = caseSensitive == null || !"false".equals(caseSensitive);
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/tasks/ArtifactArchiver.java]: NIT: I would prefer Boolean.parseString()

File: core/src/main/java/hudson/tasks/ArtifactArchiver.java [hunk 4]
```diff
@@ -294,13 +315,19 @@ public String getDisplayName() {
...
 
         /**
-         * Performs on-the-fly validation on the file mask wildcard.
+         * Performs on-the-fly validation of the file mask wildcard, when the artifacts
+         * textbox or the caseSensitive checkbox are modified
          */
-        public FormValidation doCheckArtifacts(@AncestorInPath AbstractProject project, @QueryParameter String value) throws IOException {
+        public FormValidation doCheckArtifacts(@AncestorInPath AbstractProject project,
+                @QueryParameter String value,
+                @QueryParameter(value = "caseSensitive") String caseSensitive)
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/tasks/ArtifactArchiver.java]: "value" is not required here

File: core/src/main/java/hudson/tasks/ArtifactArchiver.java (hunk 3 above)
Comment (oleg-nenashev) [inline core/src/main/java/hudson/tasks/ArtifactArchiver.java]: BTW, the input field could be boolean

**Comment by oleg-nenashev** [comment]
@dominiquebrice 
My apologies for the missing response. I've managed to miss the last ping. The PR looks good to me with the exception of a one "caseSensitive" processing. It can be done later => :+1: 

If @daniel-beck agrees, I'm going to merge the PR

**Comment by Vlatombe** [comment]
:+1:

File: core/src/main/java/hudson/tasks/ArtifactArchiver.java (hunk 3 above)
Comment (dominiquebrice) [inline core/src/main/java/hudson/tasks/ArtifactArchiver.java]: I had this approach to be very conservative regarding existing behavior, meaning I switch to NOT case sensitive only if explicitly required. So:
 - if caseSensitive is null, I want bCaseSensitive to be true (not false!)
 - if caseSensitive is "false", I want bCaseSensitive to be false
 - if caseSensitive is "true", I want bCaseSensitive to be true
 - if caseSensitive is "tutu", I want bCaseSensitive to be true (not false!)

Boolean.parseBoolean wouldn't give me that behavior.
Of course if I'm being uselessly cautious, Boolean.parseBoolean is better :-)

**Comment by dominiquebrice** [comment]
thanks for merging!

**Pull request merged**